client/ayon_core/pipeline/traits/representation.py

```diff
@@ -0,0 +1,710 @@
+            except KeyError as e:
+                msg = f"Trait with ID {trait_id} not found."
+                raise MissingTraitError(msg) from e
+
+        result = next(
+            (
+                self._data.get(trait_id)
+                for trait_id in self._data
+                if trait_id.startswith(trait_id)
+            ),
+            None,
+        )
+        if not result:
+            msg = f"Trait with ID {trait_id} not found."
+            raise MissingTraitError(msg)
+        return result
+
+    def get_traits(self,
+                     traits: Optional[list[Type[T]]]=None
+                   ) -> dict[str, T]:
+        """Get a list of traits from the representation.
+
+        If no trait IDs or traits are provided, all traits will be returned.
+
+        Args:
+            traits (list[TraitBase], optional): List of trait classes.
+
+        Returns:
+            dict: Dictionary of traits.
+
+        """
+        result: dict[str, T] = {}
+        if not traits:
+            for trait_id in self._data:
+                result[trait_id] = self.get_trait_by_id(trait_id=trait_id)
+            return result
+
+        for trait in traits:
+             result[str(trait.id)] = self.get_trait(trait=trait)
+        return result
+
+    def get_traits_by_ids(self, trait_ids: list[str]) -> dict[str, T]:
+        """Get a list of traits from the representation by their id.
+
+        If no trait IDs or traits are provided, all traits will be returned.
+
+        Args:
+            trait_ids (list[str]): List of trait IDs.
+
+        Returns:
+            dict: Dictionary of traits.
+
+        """
+        return {
+            trait_id: self.get_trait_by_id(trait_id)
+            for trait_id in trait_ids
+        }
+
+    def traits_as_dict(self) -> dict:
+        """Return the traits from Representation data as a dictionary.
+
+        Returns:
+            dict: Traits data dictionary.
+
+        """
+        return {
+            trait_id: trait.model_dump()
+            for trait_id, trait in self._data.items()
+            if trait and trait_id
+        }
+
+    def __len__(self):
+        """Return the length of the data."""
+        return len(self._data)
+
+    def __init__(
+            self,
+            name: str,
+            representation_id: Optional[str]=None,
+            traits: Optional[list[T]]=None):
+        """Initialize the data.
+
+        Args:
+            name (str): Representation name. Must be unique within instance.
+            representation_id (str, optional): Representation ID.
+            traits (list[TraitBase], optional): List of traits.
+        """
+        self.name = name
+        self.representation_id = representation_id or uuid.uuid4().hex
+        self._data = {}
+        if traits:
+            for trait in traits:
+                self.add_trait(trait)
+
+    @staticmethod
+    def _get_version_from_id(trait_id: str) -> Union[int, None]:
```

Comment:
`Union` has same issue as `list` and `dict` typehints, it was introduced later than what we do support.
```suggestion
    def _get_version_from_id(trait_id: str) -> "Union[int, None]":
```

Reply:
Or maybe I'm wrong? `Union` is available since Python 3.8 and using `list` and `dict` is available since Python 3.9 . I think we still expect 3.8 to work?

Reply:
it works because of `__future__.annotations`